docs/components/message-factory.md

```diff
@@ -0,0 +1,44 @@
+# Message Factory
+
+**Factory interfaces for PSR-7 HTTP Message.**
+
+
+## Rationale
+
+While it should be possible to use every PSR-7 aware HTTP client with any RequestInterface implementation,
+creating the request objects will still tie an application to a specific implementation.
```

Comment:
s/an application/the code/ ?

we have to work hard on consistent wording: application is the thing the end user writes. the largest issue right here however is third party libraries.